src/main/engine/GfxController/headers/GfxController.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file GfxController.hpp
+ * @author Christian Galvez
+ * @brief 
+ * @version 0.1
+ * @date 2024-01-21
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+#include <string>
+#include <vector>
+#include <Polygon.hpp>
+
+#define GFX_OK(gType) GfxResult<gType>(GfxApiResult::OK, 0);
+#define GFX_FAILURE(gType) GfxResult<gType>(GfxApiResult::FAILURE, -1);
```

Comment:
This interface should make it a lot easier to switch between different graphics backends in the future. It's a mess right now because I don't really know what kinds of calls to expect from other backends. Hopefully this class can be refactored as support for additional graphics backends is added. 

Right now, this interface is heavily tied to studious, but I'm not sure if this is the best choice. Methods like `generateVertexBuffer` make it really easy to generate vertex buffers for Polygons, but having a tight coupling to the Polygon object may end up being an anti-pattern. As more functionality is demanded of the `generateVertexBuffer` method, the Polygon class may become unnecessarily large.

These convenience methods will most likely be removed later on. The `render` method seemed like a nice method to have at first, but it's proving problematic to use across different SceneObject types.